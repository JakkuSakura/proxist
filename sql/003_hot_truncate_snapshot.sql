-- Reset tables
DROP TABLE IF EXISTS proxist.hot_ticks;
DROP TABLE IF EXISTS proxist.ticks_persist;

CREATE TABLE proxist.hot_ticks
(
    tenant String,
    symbol String,
    ts DateTime64(6),
    px Float64
) ENGINE = Memory;

CREATE TABLE proxist.ticks_persist
(
    tenant String,
    symbol String,
    ts DateTime64(6),
    px Float64
) ENGINE = MergeTree
ORDER BY (tenant, symbol, ts);

-- Seed and persist
INSERT INTO proxist.hot_ticks VALUES ('alpha', 'AAPL', toDateTime64('2024-01-01 09:30:00', 6), 100.00);
INSERT INTO proxist.hot_ticks VALUES ('alpha', 'AAPL', toDateTime64('2024-01-01 09:30:01', 6), 100.50);
INSERT INTO proxist.hot_ticks VALUES ('alpha', 'MSFT', toDateTime64('2024-01-01 09:30:02', 6), 99.75);

INSERT INTO proxist.ticks_persist SELECT * FROM proxist.hot_ticks;

-- Clear the hot cache to simulate snapshot rollover
TRUNCATE TABLE proxist.hot_ticks;

SELECT 'hot' AS layer, count() AS rows
FROM proxist.hot_ticks
UNION ALL
SELECT 'persisted' AS layer, count() AS rows
FROM proxist.ticks_persist
ORDER BY layer
FORMAT TSVWithNames;
