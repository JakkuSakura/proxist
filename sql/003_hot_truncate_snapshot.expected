tenant	symbol	total_rows	first_ts_micros	last_ts_micros
alpha	AAPL	4	1704103200000000	1704103203000000
alpha	MSFT	2	1704103201000000	1704103204000000
beta	GOOG	2	1704103203000000	1704103205000000
group_key	entity_key	route_key	memory_rows	memory_first_micros	memory_last_micros	durable_through_micros	wal_high_micros
alpha	AAPL	alpha::AAPL	4	1704103200000000	1704103203000000	1704103203000000	1704103203000000
alpha	MSFT	alpha::MSFT	2	1704103201000000	1704103204000000	1704103204000000	1704103204000000
beta	GOOG	beta::GOOG	2	1704103203000000	1704103205000000	1704103205000000	1704103205000000
